### ISSUE TITLE
add bounding boxes for subtree trivial reject

### ISSUE THREAD

**Issue by pieper** [pieper/cvg#3]
Currently every leaf is evaluated at every point in the raster.  While it's not true in general, many expressions are have a limited spatial range that can be combined.

**Comment by bebbi** [comment]
yes, bounding box is nice. Probably box/sphere would be good start. I was thinking of this too, because it might help with auto-setting a good default for raster grid boundaries (full model).

**Comment by pieper** [comment]
Yes, absolutely.
